### PR TITLE
fix: [ANDROAPP-3555] Radiobutton misbehaviour

### PR DESCRIPTION
**Pull request by Balcan**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3555)

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/res/layout/form_radio_button_horizontal.xml [hunk 1]
```diff
@@ -6,6 +6,7 @@
     <data>
 
         <import type="android.view.View" />
+        <import type="org.dhis2.R"/>
```
Comment (andresmr) [inline app/src/main/res/layout/form_radio_button_horizontal.xml:9]: Just curiosity, why do we need this?

File: app/src/main/java/org/dhis2/Bindings/Bindings.java [hunk 2]
```diff
@@ -648,4 +650,23 @@ public static void requestFocus(EditText editText, boolean focused) {
             editText.clearFocus();
         }
     }
+
+    @BindingAdapter("checkListener")
+    public static void checkListener(RadioGroup radioGroup, RadioButtonViewModel viewModel){
+        radioGroup.setOnCheckedChangeListener(null);
```
Comment (andresmr) [inline app/src/main/java/org/dhis2/Bindings/Bindings.java:656]: do you need to disable the listener, then setup with the viewModel and then setup the listener?

Comment (Balcan) [reply app/src/main/java/org/dhis2/Bindings/Bindings.java:656]: Yes.
You need to disable the listener every time a new viewModel is bound. If not, when setting the initial value the processor is triggered.

File: app/src/main/res/layout/form_radio_button_horizontal.xml (hunk 1 above)
Comment (Balcan) [reply app/src/main/res/layout/form_radio_button_horizontal.xml:9]: Not needed. I forgot to remove it

**Review by mmmateos — APPROVED** [review]

**Review by ferdyrod — APPROVED** [review]

**Pull request merged**